### ISSUE TITLE
Record processing order

### ISSUE THREAD

**Issue by orient19** [octodns/octodns-gcore#7]
Hello! I think you first need to delete records, and then create them so that there is no such conflict (look at the log).

```
octodns-sync --version
octoDNS 0.9.17
```

log:
```
********************************************************************************
* mydomain.ru.
********************************************************************************
* edgecenter (GCoreProvider)
*   Delete <CnameRecord CNAME 600, auth.mydomain.ru., fe-synterra-https2.mydomain.ru.>
*   Create <ARecord A 600, auth.mydomain.ru., ['xxx.xxx.xxx.xxx']> (config)
*   Summary: Creates=1, Updates=0, Deletes=1, Existing Records=617
********************************************************************************


2022-05-26T10:49:12  [140690960328512] INFO  GCoreProvider[edgecenter] apply: making 2 changes to mydomain.ru.
2022-05-26T10:49:12  [140690960328512] INFO  GCoreProvider[edgecenter] creating: Create <ARecord A 600, auth.mydomain.ru., ['xxx.xxx.xxx.xxx']> (config)
Traceback (most recent call last):
  File "/usr/local/bin/octodns-sync", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/octodns/cmds/sync.py", line 38, in main
    manager.sync(eligible_zones=args.zone, eligible_sources=args.source,
  File "/usr/local/lib/python3.8/dist-packages/octodns/manager.py", line 523, in sync
    total_changes += target.apply(plan)
  File "/usr/local/lib/python3.8/dist-packages/octodns/provider/base.py", line 226, in apply
    self._apply(plan)
  File "/usr/local/lib/python3.8/dist-packages/octodns_gcore/__init__.py", line 594, in _apply
    getattr(self, f"_apply_{class_name.lower()}")(change)
  File "/usr/local/lib/python3.8/dist-packages/octodns_gcore/__init__.py", line 556, in _apply_create
    self._client.record_create(
  File "/usr/local/lib/python3.8/dist-packages/octodns_gcore/__init__.py", line 111, in record_create
    self._request(
  File "/usr/local/lib/python3.8/dist-packages/octodns_gcore/__init__.py", line 88, in _request
    r.raise_for_status()
  File "/usr/local/lib/python3.8/dist-packages/requests/models.py", line 960, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 409 Client Error: Conflict for url: https://dnsapi.gcorelabs.com/v2/zones/mydomain.ru/auth.mydomain.ru./A
```

**Comment by orient19** [comment]
Hi, thanks for looking at my issue. Just in case, I'm attaching the debug output:
```
********************************************************************************
* mydomain.ru.
********************************************************************************
* edgecenter (GCoreProvider)
*   Delete <CnameRecord CNAME 600, auth.mydomain.ru., fe-synterra-https2.mydomain.ru.>
*   Create <ARecord A 600, auth.mydomain.ru., ['xxx.xxx.xxx.xxx']> (config)
*   Summary: Creates=1, Updates=0, Deletes=1, Existing Records=617
********************************************************************************


2022-05-27T09:56:56  [140624803215168] DEBUG Manager sync:   checking safety
2022-05-27T09:56:56  [140624803215168] DEBUG Manager sync:   applying
2022-05-27T09:56:56  [140624803215168] INFO  GCoreProvider[edgecenter] apply: making 2 changes to mydomain.ru.
2022-05-27T09:56:56  [140624803215168] DEBUG GCoreProvider[edgecenter] _apply: zone=mydomain.ru., len(changes)=2
2022-05-27T09:56:57  [140624803215168] DEBUG urllib3.connectionpool https://dnsapi.gcorelabs.com:443 "GET /v2/zones/mydomain.ru HTTP/1.1" 200 None
2022-05-27T09:56:57  [140624803215168] INFO  GCoreProvider[edgecenter] creating: Create <ARecord A 600, auth.mydomain.ru., ['xxx.xxx.xxx.xxx']> (config)
2022-05-27T09:56:57  [140624803215168] DEBUG urllib3.connectionpool https://dnsapi.gcorelabs.com:443 "POST /v2/zones/mydomain.ru/auth.mydomain.ru./A HTTP/1.1" 409 80
Traceback (most recent call last):
  File "/usr/local/bin/octodns-sync", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/octodns/cmds/sync.py", line 38, in main
    manager.sync(eligible_zones=args.zone, eligible_sources=args.source,
  File "/usr/local/lib/python3.8/dist-packages/octodns/manager.py", line 523, in sync
    total_changes += target.apply(plan)
  File "/usr/local/lib/python3.8/dist-packages/octodns/provider/base.py", line 226, in apply
    self._apply(plan)
  File "/usr/local/lib/python3.8/dist-packages/octodns_gcore/__init__.py", line 594, in _apply
    getattr(self, f"_apply_{class_name.lower()}")(change)
  File "/usr/local/lib/python3.8/dist-packages/octodns_gcore/__init__.py", line 556, in _apply_create
    self._client.record_create(
  File "/usr/local/lib/python3.8/dist-packages/octodns_gcore/__init__.py", line 111, in record_create
    self._request(
  File "/usr/local/lib/python3.8/dist-packages/octodns_gcore/__init__.py", line 88, in _request
    r.raise_for_status()
  File "/usr/local/lib/python3.8/dist-packages/requests/models.py", line 960, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 409 Client Error: Conflict for url: https://dnsapi.gcorelabs.com/v2/zones/mydomain.ru/auth.mydomain.ru./A
```
The diagnosis and solution of the problem remain the [same](https://github.com/octodns/octodns-gcore/issues/7#issuecomment-1138664888)?=)

**Comment by orient19** [comment]
@ross Thanks for the fix! The octodns-gcore v0.0.2 worked correctly.
```
********************************************************************************
* mydomain.ru.
********************************************************************************
* edgecenter (GCoreProvider)
*   Delete <CnameRecord CNAME 600, auth.mydomain.ru., fe-synterra-https2.mydomain.ru.>
*   Create <ARecord A 600, auth.mydomain.ru., ['xxx.xxx.xxx.xxx']> (config)
*   Summary: Creates=1, Updates=0, Deletes=1, Existing Records=617
********************************************************************************


2022-05-30T10:16:28  [140419840243520] INFO  GCoreProvider[edgecenter] apply: making 2 changes to mydomain.ru.
2022-05-30T10:16:28  [140419840243520] INFO  GCoreProvider[edgecenter] deleting: Delete <CnameRecord CNAME 600, auth.mydomain.ru., fe-synterra-https2.mydomain.ru.>
2022-05-30T10:16:29  [140419840243520] INFO  GCoreProvider[edgecenter] creating: Create <ARecord A 600, auth.mydomain.ru., ['xxx.xxx.xxx.xxx']> (config)
2022-05-30T10:16:29  [140419840243520] INFO  Manager sync:   2 total changes
```
